prov/verbs/src/verbs_cm.c

```diff
@@ -332,6 +332,10 @@ vrb_msg_ep_reject(struct fid_pep *pep, fid_t handle,
 	if (ret)
 		VRB_WARN_ERR(FI_LOG_EP_CTRL, "rdma_reject", ret);
 
+	ret = rdma_destroy_id(connreq->id) ? -errno : 0;
```

Comment:
Move this code to after rdma_reject call above, inside if (connreq->id) check.  Also assign null to connreq->id after destroy.

Reply:
Moving it under the `else if (connreq->id)` would mean we won't call `rdma_destroy_id` on `if (connreq->is_xrc)` path which also calls `rdma_reject` and then frees the `connreq` structure leaving `id` unreleased. Is that what you had in mind:

```c
	ofi_mutex_lock(&_pep->eq->event_lock);
	if (connreq->is_xrc) {
		ret = vrb_msg_xrc_ep_reject(connreq, cm_hdr,
				(uint8_t)(sizeof(*cm_hdr) + paramlen));
	} else if (connreq->id) {
		ret = rdma_reject(connreq->id, cm_hdr,
			(uint8_t)(sizeof(*cm_hdr) + paramlen)) ? -errno : 0;
		ret = rdma_destroy_id(connreq->id) ? -errno : 0;
		connreq->id = NULL;
	} else {
		ret = -FI_EBUSY;
	}
	ofi_mutex_unlock(&_pep->eq->event_lock);

	if (ret)
		VRB_WARN_ERR(FI_LOG_EP_CTRL, "rdma_reject", ret);

	free(connreq);
	return ret;
```

Reply:
Thanks for suggestions. I can see that `connreq->id` can only be NULL since de5b92d00eec91338d451e0a55818b3b0a2abaa5 for XRC endpoint hence we elide the check for NULL.